### ISSUE TITLE
[Perf] Serialize CSS as string for native shadow DOM and light DOM

### ISSUE THREAD

**Issue by nolanlawson** [salesforce/lwc#2368]
In every case except synthetic shadow DOM, we can serialize the CSS in `@lwc/style-compiler` as a simple string, rather than a function that takes the scoping tokens as input and outputs a string. This would result in better performance on the client side since we wouldn't be concatenating strings at runtime.

Cases where we can do this:

- native shadow (CSS is untransformed)
- light DOM global CSS (CSS is untransformed)
- light DOM scoped CSS (transformed once at compile time)

Here is where we currently serialize this: 

https://github.com/salesforce/lwc/blob/d93e2963d8d32b138ce9ed130d98e7bef84b3d74/packages/%40lwc/style-compiler/src/serialize.ts#L60-L62

**Comment by pmdartus** [comment]
Today there is no indication if a component is using synthetic shadow DOM or not a compile-time.
In order to do this, we would need to add a compile-time flag indicating if the compiled LWC component might rely on synthetic shadow DOM at runtime or not.

**Comment by nolanlawson** [comment]
Fair enough. This fix might have to wait until after we remove synthetic shadow DOM entirely.
